### ISSUE TITLE
React Mode // Wrong type in dropdown list in React Mode

### ISSUE THREAD

**Issue by schmupu** [ioBroker/ioBroker.admin#798]
**Describe the bug**  

if you use in common.states a dropdown list with type common.type boolean or number the value off the state is a string and not from type boolean or number.  

Value of lupusec.0.devices.ZS:a61d01.status_ex is "false" or "true". Expected value of lupusec.0.devices.ZS:a61d01.status_ex would be  false or true

```
{
  "from": "system.adapter.lupusec.0",
  "user": "system.user.admin",
  "ts": 1619900854709,
  "common": {
    "name": "Funksteckdose Haustür (Switch)",
    "type": "boolean",
    "role": "switch.power",
    "states": "false:off;true:on",
    "read": true,
    "write": true,
    "icon": "/icons/48.png"
  },
  "native": {},
  "acl": {
    "object": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator",
    "state": 1636
  },
  "_id": "lupusec.0.devices.ZS:a61d01.status_ex",
  "type": "state"
}
```

**Versions:**  
 - Adapter version: 5.0.11
 - JS-Controller version: 3.3.4
 - Node version: 12.6..3
 - Operating system: MacOS and Linux

**Comment by Apollon77** [comment]
Hm ... according specs "states" is ONLY supported on number fields which also meas that the "id" can only be numbers ...

**Comment by schmupu** [comment]
Without React mode, in the old mode it works without problems

**Comment by Apollon77** [comment]
old admin ignored that is normally not allowed :-)

**Comment by schmupu** [comment]
numbers does not work either! You get a string back (wrong format) if you choose a value from dropdown list.

```
{
  "type": "state",
  "common": {
    "type": "number",
    "role": "switch.power",
    "read": true,
    "write": true,
    "name": "Funksteckdose Haustür (Switch)",
    "states": "0:off;1:on",
    "icon": "/icons/48.png"
  },
  "native": {},
  "from": "system.adapter.lupusec.0",
  "user": "system.user.admin",
  "ts": 1619933016373,
  "_id": "lupusec.0.devices.ZS:a61d01.status_ex",
  "acl": {
    "object": 1636,
    "state": 1636,
    "owner": "system.user.admin",
    "ownerGroup": "system.group.administrator"
  }
}
```
if you choose on from dropdown list, you get back lupusec.0.devices.ZS:a61d01.status_ex = "1" instead of lupusec.0.devices.ZS:a61d01.status_ex = 1. If you choose off from dropdown list, you get back lupusec.0.devices.ZS:a61d01.status_ex = "0" instead of lupusec.0.devices.ZS:a61d01.status_ex = 0.
Values are send back in wrong format and you get an info in the log, that you choose the wrong format (new in js-controller).

It would be nice, if the dropdown list will still support boolean, numbers and strings in react mode.

**Comment by Apollon77** [comment]
Ok, then @GermanBluefox needs to say something. as said: notmally this states format was officially never defined in my eyes (also that admin 4 accepted it as it seems ...

**Comment by GermanBluefox** [comment]
Fixed in 5.0.13

**Comment by schmupu** [comment]
Tested with number and boolean. It works perfect! Thanks
